src/VisualStudio/Core/Def/DocumentOutline/DocumentSymbolData.cs

```diff
@@ -2,45 +2,29 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
 using System.Collections.Immutable;
-using System.ComponentModel;
-using System.Runtime.CompilerServices;
-using Microsoft.VisualStudio.Imaging;
-using Microsoft.VisualStudio.Imaging.Interop;
 using Microsoft.VisualStudio.LanguageServer.Protocol;
 using Microsoft.VisualStudio.Text;
 
 namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline
 {
     using SymbolKind = LanguageServer.Protocol.SymbolKind;
 
-    internal sealed class DocumentSymbolData
+    internal sealed record DocumentSymbolData(
+        string Name,
+        SymbolKind SymbolKind,
+        SnapshotSpan RangeSpan,
+        SnapshotSpan SelectionRangeSpan,
```

Comment:
doc RangeSpan vs SelectionRangeSpan.  unclear what purpose each serves.

Reply:
Might be good to still add this.